spec/daru/io/importers/redis_spec.rb

```diff
@@ -0,0 +1,192 @@
+# @note
+#
+#   Custom matchers +belong_to+ and +contain_from+ have been used,
+#   as Redis doesn't necessarily insert keys in the given order. Due
+#   to this, some rows and columns might be jumbled, and there is no
+#   way to expect for an exact match while testing on RSpec. Rather,
+#   the DataFrame is tested to have the same data, in *ANY* order.
+#
+#   Signed off by @athityakumar on 08/16/2017 at 10:00PM IST
+RSpec::Matchers.define :belong_to do |expected|
+  match { |actual| (actual.to_a.uniq - expected.to_a.uniq).empty? }
+end
+
+RSpec::Matchers.define :unordered_dataframe do |expected|
+  match do |actual|
+    actual = actual.to_a.map { |x| x.data.to_a }.flatten.uniq
+    expected.map!(&:values) unless expected.first.is_a? Array
+    expected = expected.flatten.uniq
+    (actual - expected).empty?
+  end
+end
+
+RSpec.shared_examples 'redis importer' do |use_nrows=true|
+  it_behaves_like 'daru dataframe'
+  its(:data)    { is_expected.to unordered_dataframe(expected_data)   }
+  its(:ncols)   { is_expected.to eq(ncols)                            }
+  its(:index)   { is_expected.to belong_to(expected_index)            }
+  its(:vectors) { is_expected.to belong_to(expected_vectors)          }
+
+  its(:nrows) { is_expected.to eq(nrows) } if use_nrows
+end
```

Comment:
I still feel it being really wordy :( WDYT about this thing instead:
```ruby
it { 
 is_expected.to match_dataframe_unordered(DataFrame.new(
   data,
   index: index
   vectors: vectors
 ))
}
```

It will express meaning of "what's going on here" in the clearest way possible. 
And also, probably, will be useful for other importers tests.